app/api/sections.js

```diff
@@ -56,4 +93,120 @@ router.post('/:course_id/sections', requireAuthentication, async function (req,
     }
 })
 
+// get all sections within a course
+router.get('/:course_id/sections', requireAuthentication, async function (req, res, next) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const courseId = parseInt(req.params['course_id'])
+
+    const isTeacher = await checkIfTeacher(user.id, courseId)
+
+    if (isTeacher) {
+        try {
+            const sectionsInCourse = await getSectionsInCourse(courseId)
+            if (sectionsInCourse.length != 0) {     // if sections are present for this course
+                res.status(200).send(sectionsInCourse)
+            }
+            else {
+                res.status(204).send()  // no sections for this course
+            }
+        }
+        catch (e) {
+            if (e instanceof ValidationError) {
+                return res.status(400).send({error: serializeSequelizeErrors(e)})
+            }
+            else {
+                next(e)
+            }
+        }
+    }
+    else {
+        res.status(403).send({error: "Only the teacher for a course can get a section"})
+    }
+})
+
+// get a specific section, and lectures for that section
+router.get('/:course_id/sections/:section_id', requireAuthentication, async function (req, res, next) {
```

Comment:
I know this is the URL from the endpoints doc, but do we want the url to end in :section_id/lectures instead? I think that makes more sense considering the endpoints is essentially just getting lectures

Reply:
I think it's fine either way. This does give back the section info as well. Certainly only need one or the other

Reply:
From how I interpreted the doc, this endpoint is there to mainly get the section info, and the lecture stuff is an add-on to that. There's an endpoint in lectures.js that just gets all lectures for a section, so I think we should leave this url as is (we could remove the lectures part of this if we want to, though).